spring-pulsar/src/test/java/org/springframework/pulsar/core/PulsarTopicTests.java

```diff
@@ -0,0 +1,53 @@
+package org.springframework.pulsar.core;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import java.util.stream.Stream;
+import org.apache.pulsar.common.naming.TopicDomain;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class PulsarTopicTests {
+
+	@Test
```

Comment:
What happens when you add a test w/ a string name of "some-topic-name/" (that ends in a slash)?

Reply:
Yeh this crossed my mind as well. One tradeoff on relying on the formatting in `toString` for some business logic is that it is usually not clear that is the case. We can always add docs in javadocs but it is not as direct as having a separate API. In this case I would have to see the code change to make a decision. I would say go w/ whatever feels best and we can chat more about it then.

Reply:
There is a bit of validation in `getComponents`.